SimplifiedTests/NYPLAxisLicenseServiceTests.swift

```diff
@@ -0,0 +1,204 @@
+    try! NYPLAssetWriter().writeAsset(licenseData, atURL: savedLicenseURL)
+    XCTAssertTrue(FileManager.default.fileExists(atPath: savedLicenseURL.path))
+    
+    itemDownloader.downloadsTerminated = {
+      failedValidationExpectation.fulfill()
+    }
+    
+    license.didLogLicenseError = {
+      if $0 == "AxisLicenseService validation failed due to invalid vault ID in license" {
+        failedValidationExpectation.fulfill()
+      }
+    }
+    license.validateLicense()
+    
+    wait(for: [failedValidationExpectation], timeout: 10.0)
+  }
+  
+  func testLicenseWithValidKeycheckShouldBeValidated() {
+    let validationExpectation = XCTestExpectation(
+      description: "Valid license should be validated")
+    
+    let license = createLicenseService(bookVaultId: TestBook.book2.bookVaultId,
+                                       cypher: cypherReturningBook2VaultId,
+                                       isbn: TestBook.book2.isbn)
+    
+    let licenseURL = Bundle(for: NYPLAxisLicenseServiceTests.self)
+      .url(forResource: "license-\(license.isbn)", withExtension: "json")!
+    let savedLicenseURL = licenseHomeDirectory
+      .appendingPathComponent(axisKeysProvider.desiredNameForLicenseFile)
+    if FileManager.default.fileExists(atPath: savedLicenseURL.path) {
+      try! FileManager.default.removeItem(at: savedLicenseURL)
+    }
+    
+    let licenseData = try! Data(contentsOf: licenseURL)
+    try! NYPLAssetWriter().writeAsset(licenseData, atURL: savedLicenseURL)
+    XCTAssertTrue(FileManager.default.fileExists(atPath: savedLicenseURL.path))
+    
+    license.validateLicense()
+    itemDownloader.dispatchGroup.notify(queue: .global()) {
+      if self.itemDownloader.shouldContinue {
+        validationExpectation.fulfill()
+      }
+    }
+    wait(for: [validationExpectation], timeout: 10.0)
+  }
+  
+  func testLicenseShouldSaveBookInfoForLaterUse() {
+    let savedInfoExpectation = XCTestExpectation(
+      description: "Book info should be saved for later use")
+    
+    let license = createLicenseService(bookVaultId: TestBook.book1.bookVaultId,
+                                       cypher: cypherReturningBook2VaultId,
+                                       isbn: TestBook.book1.isbn)
+    
+    let designatedBookInfoURL = licenseHomeDirectory
+      .appendingPathComponent(axisKeysProvider.bookFilePathKey)
+    
+    license.saveBookInfoForFetchingLicense()
+    itemDownloader.dispatchGroup.notify(queue: .global()) {
+      if
+        let data = try? Data(contentsOf: designatedBookInfoURL),
+        let jsonObject = try? JSONSerialization.jsonObject(with: data, options: .fragmentsAllowed),
+        let json = jsonObject as? [String: String],
+        let bookVaultId = json[self.axisKeysProvider.bookVaultKey],
+        bookVaultId == TestBook.book1.bookVaultId,
+        let isbn = json[self.axisKeysProvider.isbnKey],
+        isbn == TestBook.book1.isbn
+      {
+        savedInfoExpectation.fulfill()
+      }
```

Comment:
should we do an `else XCTFail()` so that we don't wait 10 seconds for the expectation to time out? Not sure if this can apply to other tests, but if so this would speed up test execution.

It's also probably ok to reduce the timeout?

Reply:
Done.